### PR TITLE
24389, 22238 - allowable actions logic; some other updates to allowed actions

### PR DESCRIPTION
**Pull request by hfekete**
*Issue:24389* https://github.com/bcgov/entity/issues/24389
*Issue:22238* https://github.com/bcgov/entity/issues/22238

*Description of changes:*
- feature: allowable actions are displayed depending on data from backend
- tests for allowable actions
- added interfaces for cypress
- added more mocks

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-79-si5h6n2c.web.app

File: cypress/fixtures/allowable-actions/ben-active.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          "displayName": "Address Change",
+          "name": "changeOfAddress"
+        },
+        {
+          "displayName": "Director Change",
+          "name": "changeOfDirectors"
+        },
+        {
+          "displayName": "6-Month Consent to Continue Out",
+          "name": "consentContinuationOut"
```
Comment (severinbeauvais) [inline cypress/fixtures/allowable-actions/ben-active.json:46]: There is also Consent for Amalgamation Out filing and and Amalgamation Out filing that are implemented in Business Filings UI but not released. Please create a new ticket to implement those if not already done.

File: cypress/fixtures/allowable-actions/staff/bc-active.json
Comment (severinbeauvais) [inline cypress/fixtures/allowable-actions/staff/bc-active.json:1]: Why is this list different from the BEN list?

File: cypress/fixtures/allowable-actions/ben-active.json (hunk 1 above)
Comment (hfekete) [reply cypress/fixtures/allowable-actions/ben-active.json:46]: In UI If you see file: `src/components/bcros/filing/addStaffFiling/Index.vue` line 117 we supported this button even before. Additionally, from this PR we will be using API to verify if the button should be displayed or not.

File: src/stores/dashboardActions.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    if (!businessStore.currentBusiness) {
+      return ''
+    }
+    return businessStore.currentBusiness.legalType.toUpperCase() +
+      '|' +
+      businessStore.currentBusiness.state.toUpperCase()
+  })
+
+  async function fetchVisibleActions (businessType: string, businessStatus: string) {
+    interface CouldFileI {
```
Comment (BrandonSharratt) [inline src/stores/dashboardActions.ts:19]: Likely should be in it's own file

**Review by BrandonSharratt — APPROVED** [review]

File: cypress/fixtures/allowable-actions/ben-active.json (hunk 1 above)
Comment (severinbeauvais) [reply cypress/fixtures/allowable-actions/ben-active.json:46]: I see it! Thanks for the response. Resolving.

**Review by severinbeauvais — APPROVED** [review]
LGTM but please fix conflict before merging.

Also, I recommend updating the app version (in package.json) so that you can tell what version is deployed in different environments, and can therefore deduce whether your changes should be there or not. cc: @BrandonSharratt

**Pull request merged**